### PR TITLE
Add upsert search attributes example

### PR DESCRIPTION
**Pull request by vancexu**

### CODE REVIEW

File: src/main/java/com/uber/cadence/samples/hello/HelloSearchAttributes.java [hunk 1]
```diff
@@ -70,9 +68,54 @@
 
     @Override
     public String getGreeting(String name) {
+      SearchAttributes currentSearchAttributes = Workflow.getWorkflowInfo().getSearchAttributes();
+      System.out.println("Search Attributes on start: ");
```
Comment (meiliang86) [inline src/main/java/com/uber/cadence/samples/hello/HelloSearchAttributes.java]: Use Workflow.getLogger instead.

File: src/main/java/com/uber/cadence/samples/hello/HelloSearchAttributes.java [hunk 2]
```diff
@@ -70,9 +68,54 @@
...
+
+    private void printSearchAttributes(SearchAttributes searchAttributes) {
+      if (searchAttributes == null) {
+        return;
+      }
+      searchAttributes
+          .getIndexedFields()
+          .forEach(
+              (k, v) -> {
+                System.out.printf("%s: %s\n", k, getValueForKey(k, searchAttributes));
```
Comment (meiliang86) [inline src/main/java/com/uber/cadence/samples/hello/HelloSearchAttributes.java]: Same here. Use logger.

File: src/main/java/com/uber/cadence/samples/hello/HelloSearchAttributes.java (hunk 1 above)
Comment (vancexu) [reply src/main/java/com/uber/cadence/samples/hello/HelloSearchAttributes.java]: Logger output is not very clear for demo purpose (which will also contains some worker info) So I added comments to remind user to user logger in production.

**Comment by vancexu** [comment]
> The whole idea of Hello examples is to produce "Hello World". Either change example to use searchable attributes to produce one or move it out of the hello directory and call it something else.

Updated to produce one

File: src/main/java/com/uber/cadence/samples/hello/HelloSearchAttributes.java (hunk 1 above)
Comment (mfateev) [reply src/main/java/com/uber/cadence/samples/hello/HelloSearchAttributes.java]: I would remove this line

File: src/main/java/com/uber/cadence/samples/hello/HelloSearchAttributes.java [hunk 3]
```diff
@@ -109,8 +152,10 @@ public static void main(String[] args) {
         workflowClient.newWorkflowStub(
             HelloSearchAttributes.GreetingWorkflow.class, workflowOptions);
     // Execute a workflow waiting for it to complete.
-    String greeting = workflow.getGreeting("SearchAttributes");
+    String greeting = workflow.getGreeting("World");
```
Comment (mfateev) [inline src/main/java/com/uber/cadence/samples/hello/HelloSearchAttributes.java]: I would change workflow to not get any parameters and do not return anything. And compose Hello world from the values of search attributes.

**Pull request merged**